packages/nimble-components/src/combobox/index.ts

```diff
@@ -152,7 +152,7 @@ export class Combobox
         let updatedValue = next;
 
         if (this.$fastController.isConnected && this.options) {
-            const selectedIndex = this.options.findIndex(
+            const selectedIndex = this.filteredOptions.findIndex(
```

Comment:
A couple of possible issues:
- There is now an incongruity between what you're searching and the condition you use to enter the `if` as well as the arrays you are indexing below using the `selectedIndex` determined from `filteredOptions` (i.e. you are using this index when indexing `option`).
- Is this problematic for programmatic setting of `value`, where I expect `filteredOptions` may be empty? My guess is this is fine, as I think it will all ultimately resolve to `updatedValue` being set to `next` on line 166 and `selectedIndex` being set to -1, which I think is all proper.

Reply:
Once again, I spent a lot more time that I'd like trying to trace exactly what was going on, but I settled on the fact that `options` shouldn't be returning the full set of options if there's a filter and there are no valid options to pick.

Reply:
Oof, looks like a number of tests failed. Let me handle those before you review again.

Reply:
Turns out it was just one test failing for all three browsers. The "selected" class wasn't getting cleared from the option in `list` autocomplete mode when the filter stopped matching it. Fixed this by overriding `setSelectedOptions` and calling it from `selectedIndexChanged`.